pkg/download/downloader_test.go

```diff
@@ -46,17 +52,29 @@ func TestDownload_badStatusCodeFails(t *testing.T) {
 		http.StatusBadRequest,
 		http.StatusUnauthorized,
 	} {
-		_, _, err := download.Download(download.NewURLDownload(fmt.Sprintf("%s/status/%d", srv.URL, code)))
+		_, _, err := download.Download(testctx, download.NewURLDownload(fmt.Sprintf("%s/status/%d", srv.URL, code)))
```

Comment:
This looks a bit confusing. What does this test try to achieve ? It has all response codes listed. What is the scenario and what is expected status code ?

Reply:
The purpose of this test is to ensure that the error message returned is formatted correctly based on the specific error code. I didn't think it was necessary to create several small tests for this, but I can change the name of the test so its purpose is more clear